### PR TITLE
Add CI builds, add owasp profile

### PR DESCRIPTION
**Pull request by oconnormi**
### Description of the Change

Adds Jenkins pipeline support:
* Will comment on pr at beginning and end of build process
* Performs Sonar scans

### Verification Process
Run `Build Now`

### Applicable Issues

Fixes: #8 

### Remaining Tasks:
- [x] Create sonar project

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        /*
+          Restrict nightly builds to master branch, all others will be built on change only.
+          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
+        */
+        cron(BRANCH_NAME == "master" ? "H H(17-19) * * *" : "")
+    }
+    environment {
+        DISABLE_DOWNLOAD_PROGRESS_OPTS = '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+        COVERAGE_EXCLUSIONS = '**/test/**/*,**/itests/**/*,**/*Test*,**/sdk/**/*,**/*.js,**/node_modules/**/*,**/jaxb/**/*,**/wsdl/**/*,**/nces/sws/**/*,**/*.adoc,**/*.txt,**/*.xml,**/platform-solr-server-standalone/**/*'
```
Comment (paouelle) [inline Jenkinsfile]: ❓ Should this be sanitize for this repo

**Comment by shaundmorris** [comment]
build now

**Comment by oconnormi** [comment]
Trying that again :smile: 
Build Now

**Comment by oconnormi** [comment]
Gotta make a config change apparently

**Comment by oconnormi** [comment]
One more time
Build Now

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
@paouelle yeah I wasn't sure, I just added it because its what we usually have in our builds. If we don't need it then I'll take it out :smile:

**Comment by oconnormi** [comment]
@paouelle do we want sonar and do we care about building on both linux and windows?

**Comment by paouelle** [comment]
I am just wondering if we will end up adding a lot of exceptions or wasting time investigating them.

I am good with keeping static analysis as it is always a good idea to keep code clean and detect issues ahead of time ... even if some time it can be a pain 😄

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
Think it should work this time, Build Now

**Comment by oconnormi** [comment]
Sonar failed due to not being able to determine the github url though the error message makes it look like it did get the correct repo: 

```bash
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.5.0.1254:sonar (default-cli) on project codice-test: Failed to execute project builder: org.sonar.plugins.github.PullRequestProjectBuilder: Unable to parse GitHub repository name for this project. Please check configuration:

[ERROR]   * sonar.links.scm_dev: scm:git:https://github.com/codice/codice-test.git

[ERROR]   * sonar.links.scm: https://github.com/codice/codice-test

[ERROR] -> [Help 1]

[ERROR] 

```

One possibility is that the oauth token being used isn't valid for this repo. is there something we need to do for that on the github side @shaundmorris?

**Comment by oconnormi** [comment]
Trying again this time explicitly providing the repository name to the sonar scanner.

Build Now

**Comment by oconnormi** [comment]
Really close that time. Sonar stage had the `post` actions in the wrong spot. 
Build Now

**Pull request merged**